### PR TITLE
Send all metadata keys to update run metadata

### PR DESCRIPTION
**Pull request by jjanezhang**
# Send all metadata keys to update run metadata

This update is to address missing run metadata due to concurrent update run metadata updates.
1. No longer clearing the buffered metadata on flush metadata, sending all historical + new keys on every flush
2. Flushing both wandb and model init time together on `after_load`

### CODE REVIEW

File: composer/loggers/mosaicml_logger.py [hunk 1]
```diff
@@ -93,19 +93,21 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> No
...
                 run_url = callback.run_url
                 if run_url is not None:
-                    self._log_metadata({'wandb/run_url': run_url})
                     log.debug(f'Logging WandB run URL to metadata: {run_url}')
+                    after_load_metadata['wandb/run_url'] = run_url
                 else:
                     log.debug('WandB run URL not found, not logging to metadata')
-        self._flush_metadata(force_flush=True)
+        self._log_metadata(after_load_metadata)
+        self._flush_metadata()
```
Comment (siriuslee) [inline composer/loggers/mosaicml_logger.py:110]: I don't think this is required since it's called at the end of `self._log_metadata`

Comment (jjanezhang) [reply composer/loggers/mosaicml_logger.py]: Just updated, we are going to force flush at the end of after load!

**Review by dakinggg — APPROVED** [review]
Please update the PR description to reflect current state of the PR.

**Comment by jjanezhang** [comment]
Waiting on merging this because there are likely to be issues with duplicated run events if we do not delete run event keys from update run metadata

**Comment by dakinggg** [comment]
@jjanezhang can we close this PR?

**Pull request closed without merging**